### PR TITLE
Revert: deserialize stored transactions in a rayon thread

### LINKED ISSUE #4831: Revert deserializing state transactions in `rayon` threads
## Motivation

In PR #4828, we had to increase the `lightwalletd` full sync timeout to 2.5 hours.

This might be a Zebra bug, it looks like all the blocks are slow, not just the recent large blocks.
We could have introduced this bug in PR #4805.

### Things to try

Remove the `rayon` thread from transaction deserialization:
https://github.com/ZcashFoundation/zebra/blob/9b2185ad3d7417bf4bb5a570ad81c6a027aab3b7/zebra-state/src/service/finalized_state/disk_format/block.rs#L236-L246

Check if any of the `ReadRequest::Block` calls are slow enough to turn up in Zebra's logs in the lightwalletd full sync test.
If they are fast enough, we're done!

Otherwise, here are a few more things we could try...

<details>

If they are slow, try getting all the raw transaction data from the disk, then deserializing it all at once in a single `rayon` thread:
https://github.com/ZcashFoundation/zebra/blob/9b2185ad3d7417bf4bb5a570ad81c6a027aab3b7/zebra-state/src/service/finalized_state/zebra_db/block.rs#L127-L135

Check the logs for slow reads again.

Some other things we could try:
- just spawn threads to deserialize shielded transactions
    - how do we know before they are deserialized?
    - we know the serialized length, do we just want to do it for all large transactions?
- skip cryptographic validation when deserializing trusted transactions from the state
- try multiple RPC threads (see PR #4806 for details)
- just ignore the speed issues for now

</details>

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

This PR fixes a `lightwalletd` sync performance regression caused by PR #4801.

While that change was more correct, it also causes a 20% slowdown in `lightwalletd` sync performance:
- 34 minutes to get to block 1,700,000 after this PR
- 39-40 minutes to get to block 1,700,000 before this PR

Close #4831.

## Solution

Revert the rayon threads added in PR #4801.

## Review

This PR is important for some of our upcoming users, but it is not urgent.

### Reviewer Checklist

  - [x] CI passes

### CODE REVIEW

**Comment by teor2345** [comment]
I'm just doing a final test to make sure the speed increase wasn't due to the disk cache

**Comment by teor2345** [comment]
The disk cache only saved about 45 seconds (2%), so it's still worth merging this PR.

**Comment by teor2345** [comment]
@merifyio update

**Comment by teor2345** [comment]
@mergifyio update

**Comment by teor2345** [comment]
@mergifyio update

File: zebra-state/src/service/finalized_state/disk_format/block.rs [hunk 1]
```diff
@@ -231,21 +231,12 @@ impl FromDisk for Transaction {
...
-            scope.spawn_fifo(|_scope| {
-                tx = Some(bytes.as_ref().zcash_deserialize_into().expect(
-                    "deserialization format should match the serialization format used by IntoDisk",
-                ));
-            });
-        });
-
-        tx.expect("scope has already run")
+        // TODO: skip cryptography verification during transaction deserialization from storage,
+        //       or do it in a rayon thread (ideally in parallel with other transactions)
```
Comment (dconnolly) [inline zebra-state/src/service/finalized_state/disk_format/block.rs:235]: 👍

**Review by dconnolly — APPROVED** [review]
lgtm

**Comment by teor2345** [comment]
`ssh` failed to connect to Google Cloud:
> kex_exchange_identification: Connection closed by remote host

This seems like a temporary error.

https://github.com/ZcashFoundation/zebra/runs/8081095982?check_suite_focus=true#step:6:110

**Comment by teor2345** [comment]
@mergifyio refresh

**Pull request merged**